src/modules/CRUD/components/Details/Read/AttachmentView/AttachmentList/AttachmentList.js

```diff
@@ -1,35 +1,55 @@
-import React from 'react';
+import React, { useState, useCallback, useMemo } from 'react';
 import PropTypes from 'prop-types';
-import classNames from 'classnames';
+import { useParams } from 'react-router-dom';
+import classnames from 'classnames';
+import { getView } from '../../../../../services/CRUD';
 
-import Attachment from '../Attachment';
-import Picture from '../Picture';
+import AttachmentItem from '../AttachmentItem';
 
 import './styles.scss';
 
 const AttachmentList = ({
-  name,
   attachments,
   className,
+  fetchFeature,
+  name,
+  settings,
   ...props
 }) => {
-  const Component = name === 'attachments' ? Attachment : Picture;
+  const [isDeleting, setDeleting] = useState(null);
+  const { id: paramId, layer: paramLayer } = useParams();
+
+  const updateData = useCallback(async () => {
+    const { featureEndpoint } = getView(settings, paramLayer);
+    await fetchFeature(featureEndpoint, paramId);
+  }, [fetchFeature, paramId, paramLayer, settings]);
+
+  const sanitizeAttachments = useMemo(() => (
+    attachments.map(({
+      action_url: endpoint,
+      legend: label,
+      created_at: createdAt,
+      updated_at: updatedAt,
+      ...rest
+    }) => ({
+      endpoint,
+      label,
+      isLoading: rest.id === isDeleting,
+      ...rest,
+    }))
+  ), [attachments, isDeleting]);
```

Comment:
As `sanitizeAttachments` is in component scope, other (common) props may also be forward as object properties:

```suggestion
  const commonProps = { name, setDeleting, updateData };

  const sanitizeAttachments = useMemo(() => (
    attachments.map(({
      action_url: endpoint,
      legend: label,
      created_at: createdAt,
      updated_at: updatedAt,
      ...rest
    }) => ({
      endpoint,
      label,
      isLoading: rest.id === isDeleting,
      key: rest.id,
      ...commonProps,
      ...rest,
    }))
  ), [attachments, isDeleting, commonProps]);
```

then:

```jsx
        <AttachmentItem className="attachment-list__item" {...attachmentProps} />
```

Reply:
Ok fine :D

Reply:
Ok fine :D